File: Dsa-day146.cpp
class Solution {
public:
    int countPermutations(vector<int>& complexity) {
        int n = complexity.size();
        if (n == 1) return 1;
        long long c0 = complexity[0];
        for (int i = 1; i < n; ++i) {
            if (complexity[i] <= c0) return 0;
        }
        const int MOD = 1000000007;
        long long res = 1;
        for (int i = 2; i < n; ++i) {
            res = res * i % MOD;
        }
        return res;
    }
};

File: Dsa-day146.py
class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        n = len(complexity)
        if n == 1:
            return 1
        c0 = complexity[0]
        for i in range(1, n):
            if complexity[i] <= c0:
                return 0
        MOD = 10**9 + 7
        res = 1
        for i in range(2, n):
            res = res * i % MOD
        return res